plugins/woocommerce/src/RestApi/Routes/V4/ShippingZones/Method/Controller.php

```diff
@@ -0,0 +1,319 @@
+			'/' . $this->rest_base,
+			array(
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'create_item' ),
+				'permission_callback' => array( $this, 'check_permissions' ),
+				'args'                => array_merge(
+					$this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+					array(
+						'zone_id' => array(
+							'description'       => __( 'Shipping zone ID.', 'woocommerce' ),
+							'type'              => 'integer',
+							'required'          => true,
+							'sanitize_callback' => 'absint',
+							'validate_callback' => 'rest_validate_request_arg',
+						),
+					)
+				),
+			)
+		);
+
+		// PUT - Update shipping method.
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<id>[\d]+)',
+			array(
+				'methods'             => WP_REST_Server::EDITABLE,
+				'callback'            => array( $this, 'update_item' ),
+				'permission_callback' => array( $this, 'check_permissions' ),
+				'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::EDITABLE ),
+			)
+		);
+	}
+
+	/**
+	 * Check if a given request has permission to manage shipping methods.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has permission, WP_Error otherwise.
+	 */
+	public function check_permissions( $request ) {
+		if ( ! wc_shipping_enabled() ) {
+			return new WP_Error(
+				'rest_shipping_disabled',
+				__( 'Shipping is disabled.', 'woocommerce' ),
+				array( 'status' => WP_Http::SERVICE_UNAVAILABLE )
+			);
+		}
+
+		if ( ! wc_rest_check_manager_permissions( 'settings', 'edit' ) ) {
+			return $this->get_authentication_error_by_method( $request->get_method() );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Create a shipping method.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object or WP_Error.
+	 */
+	public function create_item( $request ) {
+		// Validate zone exists.
+		$zone = $this->validate_zone( $request['zone_id'] );
+		if ( is_wp_error( $zone ) ) {
+			return $zone;
+		}
+
+		// Validate method type.
+		$method_validation = $this->validate_method_type( $request['method_id'] );
+		if ( is_wp_error( $method_validation ) ) {
+			return $method_validation;
+		}
+
+		// Add the shipping method to the zone.
+		$instance_id = $zone->add_shipping_method( $request['method_id'] );
+
+		if ( ! $instance_id ) {
+			return $this->get_route_error_by_code( self::CANNOT_CREATE );
+		}
+
+		// Get the newly created method instance.
+		$method = WC_Shipping_Zones::get_shipping_method( $instance_id );
+		if ( ! $method ) {
+			return $this->get_route_error_by_code( self::CANNOT_CREATE );
+		}
+
+		// Update method settings, enabled status, and order.
+		$result = $method->update_from_api_request( $zone, $instance_id, $request->get_params() );
+		if ( is_wp_error( $result ) ) {
+			return $result;
```

Comment:
Can we delete the shipping method if there's something wrong with the update? What do you think?

Reply:
Yup makes sense. I have added the fix and the test. I didn't find a rollback way of doing this so just added a delete method on failure. Please let me know if there's a better approach here.

Reply:
Nope, that should be fine as we don't want a stale method lingering on the db that can't be used for anything. But it'll be good to check why we didn't do this in the original API, if it has a specific reason that we can't see.